src/resolve-config.js

```diff
@@ -32,7 +32,17 @@ function _resolveConfig(filePath, opts, sync) {
   const unwrapAndMerge = arr => {
     const result = arr[0];
     const editorConfigured = arr[1];
-    return mergeEditorConfig(filePath, result, editorConfigured);
+    const merged = Object.assign(
+      {},
+      editorConfigured,
+      mergeOverrides(Object.assign({}, result), filePath)
+    );
+
+    if (Object.keys(merged).length === 0) {
+      return null;
```

Comment:
I'm not sure if we should treat 0-option as not-found, but if so, we should document it on the README as well.

Reply:
Yeah, I'm thinking that we shouldn't distinguish between the cases where `.prettierrc`/`.editorconfig` are not present, and the cases where they are but none of their configurations pertain to the file in question. Given that the goal of `resolveConfig` is to find the configuration for a specific file, it doesn't seem especially useful to return an empty object.

I just clarified this in https://github.com/prettier/prettier/pull/2760/commits/d44420c8ee651ffb1dbec04ddcc22c60a58c7dd6

Reply:
I was thinking in using this difference to test for the existence of a `.prettierrc` to implement *format only when there is a config" (vscode).
- Empty bag -> use defaults
- null -> skip

@robwise May have implemented it this way (atom).

Anyway, if it also takes `editorConfig` into account, this is not what I would need anymore. Would it be possible to expose `findConfigPath` in that case?

Reply:
Yes we definitely want to keep this distinction. An empty config file is different to no config file.

Reply:
> @robwise May have implemented it this way (atom).

Right now I can't tell the difference between an empty config file and a missing one, so I have to put a special note to users that they need to have a rule in it in order for it to be picked up:


![prettier config format on save options](https://cl.ly/1w1a2t2X0808/Image%202017-10-18%20at%2012.57.19%20PM.png)

It would be nice to have a way to differentiate so we won't have this edge case.

Reply:
Thanks for the feedback, everyone! I see why being able to differentiate between these cases is useful, so I'll try to add some patches implementing/testing this.

Reply:
> I see why being able to differentiate between these cases is useful, so I'll try to add some patches implementing/testing this

I thought about this a bit more, and I'd actually like to let it be part of a separate PR later, since it's a pre-existing issue. That way, this PR can focus on the editorconfig functionality.